libcontainer/cgroups/systemd/common.go

```diff
@@ -365,22 +366,24 @@ func systemdVersion(conn *systemdDbus.Conn) (int, error) {
 			return
 		}
 
-		// verStr is like "v245.4-1.fc32" (including quotes)
-		if !strings.HasPrefix(verStr, `"v`) {
-			versionErr = fmt.Errorf("can't parse version %s", verStr)
-			return
-		}
-		// remove `"v` prefix and everything after the first dot
-		ver, err := strconv.Atoi(strings.SplitN(verStr[2:], ".", 2)[0])
-		if err != nil {
-			versionErr = err
-		}
-		version = ver
+		version, versionErr = systemdVersionAtoi(verStr)
+		return
 	})
 
 	return version, versionErr
 }
 
+func systemdVersionAtoi(verStr string) (int, error) {
+	// verStr should be of the form:
+	// "v245.4-1.fc32", "245", "v245-1.fc32", "245-1.fc32"
+	// all the input strings include quotes, and the output int should be 245
+	// thus, we unconditionally remove the `"v`
+	// and then match on the first integer we can grab
+	re := regexp.MustCompile(`[0-9]+`)
+	ver, err := strconv.Atoi(re.FindString(strings.Trim(verStr, `"v`)))
```

Comment:
Well, I guess if you end up using `regex.FindString`, you can do everything with it, i.e. there's no longer need to remove the prefix.

Or, if you want to be more strict about what you're parsing, keep the `Trim` (btw TrimPrefix will be better since we're no longer caring about the " at the end), and add `^` to the regex so it would want to see the numbers at the very beginning.

I would do the first way (leave the regex as is, remove strings.Trim as redundant).

Reply:
I went the pure regexp route, as suggested